src/components/FilterPanel.jsx

```diff
@@ -9,77 +9,284 @@ import {
   Typography,
   Button,
   Toolbar,
+  Chip,
+  Container,
+  Alert,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  DialogActions,
 } from '@mui/material';
 import useQueryParams from '../lib/useQueryParams';
 import AmountFilter from './AmountFilter';
 import GradeLevelFilter from './GradeLevelFilter';
 import MajorFilter from './MajorFilter';
 import CloseIcon from '@mui/icons-material/Close';
 import PropTypes from 'prop-types';
+import GradeLevel from '../types/GradeLevel';
 
-export default function FilterPanel({ onClose }) {
-  const [{ minAmount, maxAmount, grades, majors }, setQueryParam] =
-    useQueryParams();
+function FilterChip({ label, onDelete, variant }) {
+  return (
+    <Chip
```

Comment:
Some suggestions:
- Add a checkmark icon on the left for active filters that will still be applied after any pending changes
- For active filters that will not be applied after pending changes:
  - Gray it out/make it unclickable/undeletable
  - No checkmark icon
  
 Some properties to look into: `clickable`, `disabled`, `color`, `variant`, `onDelete`. You might change some or all of those